packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -112,6 +117,69 @@ export class LangiumGrammarValidator {
                 });
             }
         }
+        const importedRules = new Set<string>();
+        const resolvedGrammars = resolveTransitiveImports(this.documents, grammar);
+        for (const resolvedGrammar of resolvedGrammars) {
+            for (const rule of resolvedGrammar.rules) {
+                importedRules.add(rule.name.toLowerCase());
+            }
+        }
+        for (const name of ruleMap.keys()) {
+            if (importedRules.has(name)) {
+                const rules = ruleMap.get(name);
+                rules.forEach(e => {
+                    accept('error', `A rule with the name '${e.name}' already exists in an imported grammar.`, { node: e, property: 'name' });
+                });
+            }
+        }
+    }
+
+    /**
+     * Compared to the validation above, this validation only checks whether two imported grammars export the same grammar rule.
+     */
+    checkUniqueImportedRules(grammar: ast.Grammar, accept: ValidationAcceptor): void {
+        const imports = new Map<ast.GrammarImport, ast.Grammar[]>();
+        for (const imp of grammar.imports) {
+            const importedGrammars = resolveTransitiveImports(this.documents, imp);
+            imports.set(imp, importedGrammars);
+        }
+        const allDuplicates = new MultiMap<ast.GrammarImport, string>();
+        for (const outerImport of grammar.imports) {
+            const outerGrammars = imports.get(outerImport)!;
+            for (const innerImport of grammar.imports) {
+                if (outerImport === innerImport) {
```

Comment:
This does not cover the case where a grammar is imported twice. But that could be done in a separate validation.